### PR TITLE
pagsuite: init at 1.80

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

The PAGSuite contains optimization tools for the (pipelined) multiple constant multiplication ((P)MCM) problem.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/math/pagsuite/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    gmp
+    scalp
+  ];
+
+  enableParallelBuilding = false;
+
+  meta = with lib; {
+    description = "Optimization tools for the (P)MCM problem";
+    homepage = "https://gitlab.com/kumm/pagsuite";
+    platforms = platforms.unix;
```
Comment (azahi) [inline pkgs/applications/science/math/pagsuite/default.nix:34]: This is the default.
```suggestion
```

File: pkgs/applications/science/math/pagsuite/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "pagsuite";
+  version = "1.80";
+
+  src = fetchzip {
+    url = "https://gitlab.com/kumm/${pname}/-/raw/master/releases/${pname}_${builtins.replaceStrings ["."] ["_"] version}.zip";
+    stripRoot = false;
+    postFetch = ''mv $out/${pname}*/* $out'';
+    sha256 = "sha256-JuRuDPhKKBGz8jUBkZcZW5s2berOewjsPNR/n7kuofY=";
```
Comment (azahi) [inline pkgs/applications/science/math/pagsuite/default.nix:17]: Nit:
```suggestion
    sha256 = "sha256-JuRuDPhKKBGz8jUBkZcZW5s2berOewjsPNR/n7kuofY=";
    stripRoot = false;
    postFetch = ''
      mv $out/${pname}*/* $out
    '';
```

File: pkgs/applications/science/math/pagsuite/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+, cmake
+, gmp
+, scalp
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pagsuite";
+  version = "1.80";
+
+  src = fetchzip {
```
Comment (azahi) [inline pkgs/applications/science/math/pagsuite/default.nix:13]: Is there a reason to use `fetchzip` here and not `fetchFromGitLab`?

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/math/pagsuite/default.nix (hunk 3 above)
Comment (wegank) [reply pkgs/applications/science/math/pagsuite/default.nix:13]: The version was released before the transition to Git.

File: pkgs/applications/science/math/pagsuite/default.nix (hunk 1 above)
Comment (wegank) [reply pkgs/applications/science/math/pagsuite/default.nix]: Done.

File: pkgs/applications/science/math/pagsuite/default.nix (hunk 2 above)
Comment (wegank) [reply pkgs/applications/science/math/pagsuite/default.nix]: Done.

**Comment by azahi** [comment]
Result of `nixpkgs-review pr 188718` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pagsuite</li>
  </ul>
</details>

**Review by azahi — APPROVED** [review]

**Comment by azahi** [comment]
I would be nice if you could also move include files to a separate `dev` output.

File: pkgs/applications/science/math/pagsuite/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  ];
+
+  buildInputs = [
+    gmp
+    scalp
+  ];
+
+  enableParallelBuilding = false;
+
+  meta = with lib; {
```
Comment (sikmir) [inline pkgs/applications/science/math/pagsuite/default.nix:33]: `license` is required.

Comment (azahi) [reply pkgs/applications/science/math/pagsuite/default.nix:33]: The developer didn't provide the license so we must assume that it's unfree, and `unfree` is set by default in `meta`. Do we really need to explicitly specify it?

Comment (sikmir) [reply pkgs/applications/science/math/pagsuite/default.nix:33]: I see, `You can use and/or modify it for research purposes (please ask for push access to make it accessable!), for commercial use, please ask for a license.` and nothing more. OK, let's keep default.

Comment (wegank) [reply pkgs/applications/science/math/pagsuite/default.nix:33]: Thanks! Set to `unfree`.

File: pkgs/applications/science/math/pagsuite/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  buildInputs = [
+    gmp
+    scalp
+  ];
+
+  enableParallelBuilding = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/pagsuite/default.nix:31]: Can you add a short comment why we want to disable this.

File: pkgs/applications/science/math/pagsuite/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/applications/science/math/pagsuite/default.nix:13]: I can't find the release online https://gitlab.com/kumm/pagsuite/-/releases

File: pkgs/applications/science/math/pagsuite/default.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+, gmp
+, scalp
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pagsuite";
+  version = "1.80";
+
+  src = fetchzip {
+    url = "https://gitlab.com/kumm/${pname}/-/raw/master/releases/${pname}_${builtins.replaceStrings ["."] ["_"] version}.zip";
```
[the remark this answers is not in the document]
Comment (wegank) [reply pkgs/applications/science/math/pagsuite/default.nix]: Done.

File: pkgs/applications/science/math/pagsuite/default.nix (hunk 5 above)
Comment (wegank) [reply pkgs/applications/science/math/pagsuite/default.nix]: I can no longer reproduce the build failure, so I removed it.

**Pull request merged**